src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java

```diff
@@ -269,12 +264,63 @@ public void actionPerformed(ActionEvent e) {
             }
         };
         toolbarPanel.setResetSpeedListener(resetSpeedListener);
-
+        final int[] regionWidth = new int[1];
+        final int[] regionHeight = new int[1];
         regionsPanel.addComponentListener(new ComponentAdapter() {
             public void componentResized(ComponentEvent ev) {
+                regionWidth[0] = ev.getComponent().getWidth();
+                regionHeight[0] = ev.getComponent().getHeight();
                 renderRunner.notifyRegionResized(ev.getComponent().getWidth(), ev.getComponent().getHeight());
             }
         });
+        final boolean isReplayFinal = isReplay;
+        regionsPanel.addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                renderRunner.playback.isPaused = true;
```

Comment:
Pausing is just for the prototype right? The live playback can't be paused (I think). More importantly though, we want the data in the popup to render the current snapshot. The popup data should change as the data is updated.

Reply:
The live one can't be paused. I do the pausing is because right now the popup data can only send the snapshot when the mouse is clicked. I am not quite sure how to continuously send the snapshot data from the visualizer to the popup window. Therefore, I let the visualizer pause as the popup window is evoked.